modules/infrastructure/infrastructure.tf

```diff
@@ -0,0 +1,39 @@
+resource "duplocloud_infrastructure" "this" {
+  infra_name         = coalesce(var.name, terraform.workspace)
+  cloud              = local.clouds[var.cloud]
+  region             = var.region
+  enable_k8_cluster  = var.class == "k8s"
+  enable_ecs_cluster = var.class == "ecs"
+  address_prefix     = local.address_prefix
+  azcount            = var.azcount
+  subnet_cidr        = var.subnet_cidr
+  lifecycle {
+    ignore_changes = [
+      # don't try and recreate because any of these changed
+      # especialy address_prefix because it might be null and the 
+      address_prefix,
+      cloud
+    ]
+  }
+}
+
+resource "duplocloud_infrastructure_setting" "this" {
+  count      = var.settings != null ? 1 : 0
+  infra_name = local.name
+  dynamic "setting" {
+    for_each = var.settings
+    content {
+      key   = setting.key
+      value = setting.value
+    }
+  }
+}
+
+resource "duplocloud_infrastructure_subnet" "this" {
```

Comment:
just noting this resource seems to be good for aws and azure but doesn't seem like it supports gcp but I haven't verified that yet. But may be good to label that this module should needs work before it can be used beyond aws